app/models/resource_group.rb

```diff
@@ -5,6 +5,10 @@ class ResourceGroup < ApplicationRecord
   has_many :vm_or_templates
 
   # Rely on default scopes to get expected information
-  has_many :vms, :class_name => 'Vm'
+  has_many :vms, :class_name => 'Vm', :dependent => :destroy
```

Comment:
We typically only want one model to `dependent => destroy` sub-collections, typically it is the `ExtManagementSystem` which does this as that is the "primary" owner of the child records.  Usually these deletions are handled by `EmsRefresh` since that is the main source of inventory state.

In these cases we usually go for `:dependent => :nullify` to make sure that we don't have dangling references but we aren't deleting inventory out from under EmsRefresh.

Reply:
Thanks @agrare for the explanation. I modified the value of the `:dependent` option from `:destroy` to `:nullify` for all dependent models associated with a `ResourceGroup`.

 I was trying to be consistent with the rest of the codebase and checked what's in models that seemed related to grouping (for example `NetworkGroup` `ContainerGroup`, `MiddlewareServerGroup`) which used `:dependent: => :destroy`. But I see now that that there are many others using `:nullify` for the dependent models too.